### PR TITLE
Fixes a FTBFS with CMake v3.8

### PR DESCRIPTION
**Pull request by luis-pereira**
CMake v3.8 AUTOUIC changed the directory where the file is generated to:
<CMAKE_CURRENT_BINARY_DIR>/<TARGETNAME>_autogen/include. The changes makes
us fail.
The change is actually just an sympton and not a root cause. We shouldn't
include a ui generated file in public header.

Closes https://github.com/lxde/lxqt/issues/1277

### CODE REVIEW

**Comment by palinek** [comment]
GTM

File: liblxqt-config-cursor/selectwnd.cpp [hunk 1]
```diff
@@ -37,22 +38,24 @@
 
 #define HOME_ICON_DIR QDir::homePath() + "/.icons"
 
-SelectWnd::SelectWnd(LXQt::Settings* settings, QWidget *parent) : QWidget(parent), mSettings(settings)
+SelectWnd::SelectWnd(LXQt::Settings* settings, QWidget *parent)
+    : QWidget(parent),
+      mSettings(settings),
+      ui(new Ui::SelectWnd)
```
Comment (palinek) [inline liblxqt-config-cursor/selectwnd.cpp]: Hm...actually, the `Ui::SelectWnd` object gets never destroyed. We should use a smart pointer here.

**Comment by yan12125** [comment]
As now ```"${CMAKE_CURRENT_BINARY_DIR}/../liblxqt-config-cursor"``` is not needed, how about removing them?

File: liblxqt-config-cursor/selectwnd.cpp (hunk 1 above)
Comment (luis-pereira) [inline liblxqt-config-cursor/selectwnd.cpp]: Upsss
Forgot to amend the commit. I simply had this:
``` diff
diff --git a/liblxqt-config-cursor/selectwnd.cpp b/liblxqt-config-cursor/selectwnd.cpp
index ccd6312..8a1fa18 100644
--- a/liblxqt-config-cursor/selectwnd.cpp
+++ b/liblxqt-config-cursor/selectwnd.cpp
@@ -83,6 +83,7 @@ SelectWnd::SelectWnd(LXQt::Settings* settings, QWidget *parent)
 
 SelectWnd::~SelectWnd()
 {
+    delete ui;
 }
```

**Comment by luis-pereira** [comment]
@yan12125 That I didn't see. It probably can. Will test it.

**Comment by yan12125** [comment]
I guess this is ready to merge? @palinek

**Comment by palinek** [comment]
Yes, ship it, please.

**Pull request merged**